roles/openshift_health_checker/openshift_checks/sdn.py

```diff
@@ -0,0 +1,698 @@
+                # on busy systems.
+                # service_prefix = self.get_var('openshift_service_type')
+                # if self._templar is not None:
+                #     service_prefix = self._templar.template(service_prefix)
+                # self.save_service_logs('%s-node' % service_prefix)
+                self.save_command_output('bridges', ['brctl', 'show'])
+
+                self.save_command_output('docker-ps', ['docker', 'ps'])
+
+                self.save_command_output('flows', ['ovs-ofctl',
+                                                   '-O', 'OpenFlow13',
+                                                   'dump-flows', 'br0'])
+                self.save_command_output('ovs-show', ['ovs-ofctl',
+                                                      '-O', 'OpenFlow13',
+                                                      'show', 'br0'])
+
+                self.save_command_output('tc-qdisc', ['tc', 'qdisc', 'show'])
+                self.save_command_output('tc-class', ['tc', 'class', 'show'])
+                self.save_command_output('tc-filter', ['tc', 'filter', 'show'])
+
+                docker_unit = self.read_command_output(['systemctl', 'cat',
+                                                        'docker.service'])
+                self.register_file('docker-unit-file', docker_unit)
+
+                # Save docker service unit environment file for openshift-sdn,
+                # if any.
+                #
+                # The systemd and service modules both return the unit as a
+                # dict with only the last value for a given key, and there may
+                # be multiple EnvironmentFile stanzas, so we need to slurp and
+                # parse the service unit without the aid of these two modules.
+                #
+                # XXX: Do we really need to look for an 'openshift-sdn'
+                # environment file? It is not evident that the installer ever
+                # configures one.
+                #
+                # TODO: Submit a PR to fix the systemd module and maybe the
+                # service module as well to return a list value for any key
+                # that is repeated in the unit.
+                #
+                # TODO: Submit a PR to fix the ConfigParser Python module to
+                # support repeated keys
+                # (cf. <https://bugs.python.org/issue31528>).
+                for line in docker_unit.splitlines():
+                    if '=' not in line:
+                        continue
+
+                    key, value = line.split('=', 1)
+                    if key == 'EnvironmentFile' and 'openshift-sdn' in value:
+                        self.register_file('docker-network-file', value)
+            except OpenShiftCheckException as exc:
+                self.register_failure(exc)
+
+        self.process_pods_for_node(node_name)
+
+        # Choose a pod to use as the base pod for connectivity checks.
```

Comment:
going back to the "use case" thing again, I suspect we probably don't need this: if they can actually create pods they should use `oc adm diagnostics`. (I think...)